requirements_dev.txt

```diff
@@ -1,15 +1,40 @@
-pip==18.1
+pip==20.0.2
 bumpversion==0.5.3
-wheel==0.32.1
-watchdog==0.9.0
-flake8==3.5.0
-tox==3.5.2
-coverage==4.5.1
-Sphinx==1.8.1
-twine==1.12.1
-
-pytest==3.8.2
-pytest-runner==4.2
-pre-commit==1.17.0
-virtualenv==16.1.0
-typing_extensions>=3.7
+wheel==0.34.2
+watchdog==0.10.1
+flake8==3.7.9
+tox==3.14.3
+coverage==5.0.3
+Sphinx==2.3.1
+twine==3.1.1
+
+pre-commit==2.2.0
+virtualenv==16.7.9
+
+plotly==4.5.0
+attrs==19.3.0
+certifi==2019.11.28
+chardet==3.0.4
+idna==2.9
+importlib-metadata==1.6.0
+more-itertools==8.2.0
+numpy==1.18.2
+packaging==20.1
+pandas==1.0.3
+pluggy==0.13.1
+py==1.8.1
+pyparsing==2.4.6
+pytest==5.3.5
+pytest-runner==5.2
+python-dateutil==2.8.1
+pytz==2019.3
+requests==2.23.0
+six==1.14.0
+typing-extensions==3.7.4.1
+urllib3==1.25.8
+wcwidth==0.1.8
+zipp==3.0.0
+
+jupyter==1.0.0
+matplotlib==3.2.1
```

Comment:
I kind of  don't remember how it became so big requirements file.
Probably I did a pip freeze?
However it seems everything is here, like the test related requirements could be in requirements-test.txt

Reply:
@KonradUdoHannes : I started a Tutorial environment specification. I guess you want to keep the `requirements_dev.txt` as is for continous integration tools?
@elekt : I created already a "tutorial" conda-environment specification in one of my branches. So I would rather rename it as `requirements_tutorial.txt`

Reply:
@enryH, @elekt one important aspekt is, that the requirements file stays consistent with the build package, otherwise it gets confusing. The requirements for the build package are given in the setup.py and are pandas>=0.25, requests and typing_extensions + pytest/pytest-runner for testing. Everything else is an implicit dependence. This seems to create an issue with the pandas version as we are now using 1.0 funcitonality.

Other than that I like the idea of having additonal requirements files for differen purpouses. `_dev` was intended for the dev tools like pre-commit, tox, twine etc. Having requirements for ues cases/tutorials seems like a very good idea, particularly because some of the packages appearing in certain use cases are quite large and are not be needed for basic installations or dev.

Reply:
@KonradUdoHannes  
So what is the course of action here?
Revert this file to the previous entries but updated versions where suitable?
Then separate it for  -dev, -test, and possibly -use_cases files?

Reply:
I think separate files reasonable, possibly including a combined file for convenience. Testing I consider as part of the package itself so I don't thing a separate file is needed.
We wanted to make a small package update soon anyways in order to incorporate the unit meta data. I think in that context we can simply make sure that the requirements files are consistent with the latest setup.py.